includes/data/loader/class-wc-db-loader.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * DataLoader - WC_Db_Loader
+ *
+ * Loads Models for WooCommerce objects defined in custom DB tables.
+ *
+ * @package WPGraphQL\WooCommerce\Data\Loader
+ * @since 0.5.0
+ */
+
+namespace WPGraphQL\WooCommerce\Data\Loader;
+
+use GraphQL\Deferred;
+use GraphQL\Error\UserError;
+use WPGraphQL\Data\Loader\AbstractDataLoader;
+use WPGraphQL\WooCommerce\Data\Factory;
+use WPGraphQL\WooCommerce\Model\Tax_Rate;
+
+/**
+ * Class WC_Db_Loader
+ */
+class WC_Db_Loader extends AbstractDataLoader {
+	/**
+	 * Stores loaded CPTs.
+	 *
+	 * @var array
+	 */
+	protected $loaded_objects;
+
+	/**
+	 * Loader type
+	 * 
+	 * @var string
+	 */
+	protected $loader_type;
+	
+	/**
+	 * WC_Db_Loader constructor
+	 * 
+	 * @param AppContext $context      AppContext instance.
+	 * @param string     $loader_type  Type of loader be initialized.
+	 */
+	public function __construct( $context, $loader_type ) {
+		$this->loader_type = $loader_type;
+		parent::__construct( $context );
+	}
+
+    /**
+	 * Given array of keys, loads and returns a map consisting of keys from `keys` array and loaded
+	 * posts as the values
+	 *
+	 * @param array $keys - array of IDs.
+	 *
+	 * @return array
+	 * @throws \Exception Invalid loader type
+	 */
+	public function loadKeys( array $keys ) {
+		$loader = null;
+		switch( $this->loader_type ) {
+			case 'CART_ITEM':
+				$loader = array( $this, 'load_cart_item_from_key' );
+				break;
+			case 'DOWNLOADABLE_ITEM':
+				$loader = array( $this, 'load_downloadable_item_from_id' );
+				break;
+			case 'TAX_RATE':
+				$loader = array( $this, 'load_tax_rate_from_id' );
+				break;
+			default:
+				$loader = apply_filters( 'woographql_db_loader_func', null );
+				if ( empty( $loader ) ) {
+					throw new \Exception(
+						/* translators: %s: Loader Type */
+						sprintf( __( 'Loader type invalid: %s', 'wp-graphql-woocommerce' ), $this->loader_type )
+					);
+				}
+		}
+
+		$loaded_items = array();
+
+		/**
+		 * Loop over the keys and return an array of cart items,
+		 */
+		foreach ( $keys as $key ) {
+			$loaded_items[ $key ] = call_user_func( $loader, $key );
+		}
+
+		return ! empty( $loaded_items ) ? $loaded_items : array();
+	}
+	
+	/**
+	 * Returns the cart item connected the provided key.
+	 * 
+	 * @param string $key - Cart item key.
+	 *
+	 * @return array
+	 */
+	public function load_cart_item_from_key( $key ) {
+		// Add the cart item's product and product variation to WC-CPT buffer.
+		return Factory::resolve_cart()->get_cart_item( $key );
+	}
+
+	/**
+	 * Returns the downloadable item connected the provided IDs.
+	 * 
+	 * @param int $id - Downloadable item ID.
+	 * 
+	 * @return WC_Customer_Download|null
+	 */
+	public function load_downloadable_item_from_id( $id ) {
+		$node = new \WC_Customer_Download( $id );
```

Comment:
I'd suggest importing this class to the top of the file for better visibility of the classes being used inside this file/class.

Reply:
Maybe. I'm completely dead set on this method of resolution of the `DownloadableItem` type. Seeing as it actually doesn't work for `DownloadableItem` connections, I actually bypass this method by **priming** the objects returned from the **$source** downloadable items getters define by WC on respective crud object [here](https://github.com/wp-graphql/wp-graphql-woocommerce/pull/249/files#diff-f6fe31218f292074ac3690e67c033ecbR146-R148).